pkg/store/limiter.go

```diff
@@ -123,10 +125,31 @@ type limitedStoreServer struct {
 	newSeriesLimiter      SeriesLimiterFactory
 	newSamplesLimiter     ChunksLimiterFactory
 	failedRequestsCounter *prometheus.CounterVec
+
+	// This is a read-only field once it's set.
+	// Value 0 disables the feature.
+	maxPendingRequests        int32
+	pendingRequests           atomic.Int32
+	maxPendingRequestLimitHit prometheus.Counter
```

Comment:
should we consider track this for all grpc API with `grpc_method`?

Reply:
This limiter only intercepts `series()` calls. Only write()/series() are called heavily. Not worth the effort to track other methods.